tests/tst_ncio_mpi.F90

```diff
@@ -0,0 +1,187 @@
+! This is a test for the NCEPLIBS-ncio package.
+program tst_ncio
+
+  use mpi
+  use netcdf
+  use module_ncio
+  implicit none
+
+  character(len=72) charatt, time_units
+  type(Dataset) :: dset, dsetin
+  type(Variable) :: var
+  real(4), allocatable, dimension(:) :: values_1d
+  real(4), allocatable, dimension(:,:) :: values_2d
+  real(4), allocatable, dimension(:,:,:) :: values_3d
+  real(4), allocatable, dimension(:,:,:,:) :: values_4d
+  real(4), dimension(10,10) :: quantize1, quantize2
+  real(4) mval,r4val,qerr
+  integer ndim,nvar,ndims,ival,idate(6),icheck(6),ierr,n,nbits
+  integer :: my_rank, nprocs
+  integer :: mpi_err
+  logical hasit
+
+  call mpi_init(mpi_err)
+  call MPI_Comm_rank(MPI_COMM_WORLD, my_rank, mpi_err)
+  call MPI_Comm_size(MPI_COMM_WORLD, nprocs, mpi_err)
+
+  if (my_rank .eq. 0) print*, '*** Testing NCEPLIBS-ncio.'
+  dsetin = open_dataset('dynf000_template.nc.in')
+
+  if (my_rank .eq. 0) print *,'*** Test creation of new dataset from template...'
+  dset = create_dataset('dynf000.nc',dsetin)
+
+  if (my_rank .eq. 0) print *,'*** Test that number of variables,dimensions,attributes is read...'
+  if (dsetin%nvars .ne. 23) stop 4
+
+  if (dsetin%ndims .ne. 5) stop 5
+
+  if (dsetin%natts .ne. 8) stop 6
+
+  call close_dataset(dsetin)
+
+  if (my_rank .eq. 0) print *,'*** Test read of variable data...'
+  call read_vardata(dset, 'pressfc', values_3d)
+  call read_vardata(dset, 'vgrd', values_4d)
+  values_3d=1.013e5
```

Comment:
You read the values, but don't check them for correctness.

Reply:
This was fixed in a commit